src/__tests__/actions/loginAction/__test__/loginAction.test.js

```diff
@@ -0,0 +1,85 @@
+import configureMockStore from "redux-mock-store";
+import thunk from "redux-thunk";
+import fetchMock from "fetch-mock";
+import loginAction from "../../../../actions/loginAction";
+import { LOGIN_SUCCESS, LOGIN_FAIL } from "../../../../actions/types";
+
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+const { BASE_URL } = process.env;
+const loginUrl = `${BASE_URL}/users/login/`;
+
+describe("login actions", () => {
+  afterEach(() => {
+    fetchMock.restore();
+  });
+
+  it("handles LOGIN_SUCCESS action after logging in", () => {
+    fetchMock.postOnce(loginUrl, {
+      body: { user: { token: "abdjakbd3jkbhdkkb3hjkbjdbjdjj" } },
+      headers: { "content-type": "application/json" },
+    });
+
+    const expectedAction = [{
+      type: LOGIN_SUCCESS,
+      payload: "abdjakbd3jkbhdkkb3hjkbjdbjdjj",
+    }];
+
+    const validData = {
+      email: "testuser@gmail.com",
+      password: "testing123",
+    };
+
+    const store = mockStore();
+    
+
+    return store.dispatch(loginAction(validData)).then(() => {
+      expect(store.getActions()).toEqual(expectedAction);
+    });
+  });
+
+  it("handles LOGIN_FAIL action after logging in", () => {
+    fetchMock.postOnce(loginUrl, {
+      body: { errors: {} },
+      headers: { "content-type": "application/json" },
+    });
+
+    const expectedAction = [{
+      type: LOGIN_FAIL,
+      payload: {},
+    }];
+
+    const invalidData = {
+      email: "",
+      password: "",
+    };
+
+    const store = mockStore();
```

Comment:
You can declare one constant store in the describe and reuse it elsewhere. Instead of declaring it in every `it`

Reply:
Thank you i have moved it to describe